subsys/bluetooth/controller/ll_sw/ull_peripheral_iso.c

```diff
@@ -35,3 +80,200 @@ int ull_peripheral_iso_reset(void)
 {
 	return 0;
 }
+
+uint8_t ull_peripheral_iso_acquire(struct ll_conn *acl,
+				   struct pdu_data_llctrl_cis_req *req,
+				   uint16_t *cis_handle)
+{
+	struct ll_conn_iso_group *cig;
+	struct ll_conn_iso_stream *cis;
+
+	/* Get CIG by id */
+	cig = ll_conn_iso_group_get_by_id(req->cig_id);
+	if (!cig) {
+		/* CIG does not exist - create it */
+		cig = ll_conn_iso_group_acquire();
+
+		memset(&cig->lll, 0, sizeof(cig->lll));
+
+		cig->cig_id = req->cig_id;
+		cig->lll.handle = 0xFFFF;
+		cig->lll.role = acl->lll.role;
+
+		for (int i = 0; i < CONFIG_BT_CTLR_CONN_ISO_STREAMS_PER_GROUP;
+		     i++) {
+			cig->cis_handles[i] = 0xFFFF;
+			cig->lll.cis_handles[cig->lll.num_cis] = 0xFFFF;
+		}
+
+		ull_hdr_init(&cig->ull);
+		lll_hdr_init(&cig->lll, cig);
```

Comment:
Tis block seems fairly generic for central and peripheral roles. Should it be factored out into ull_conn_iso.c?

Reply:
> Tis block seems fairly generic for central and peripheral roles. Should it be factored out into ull_conn_iso.c?

I vote for "later". When implementing central we will have better overview of code-reuse.